client/src/components/advancedSelectWidget/AdvancedSelect.js

```diff
@@ -275,19 +277,20 @@ const AdvancedSelect = ({
       fetchResultsDebounced(searchTerms)
     }
   }, [fetchType, fetchResults, fetchResultsDebounced, searchTerms])
-
   useEffect(() => {
     if (doReset) {
       setIsLoading(false)
       setShowOverlay(false)
       setFilterType(firstFilter)
-      setSearchTerms(selectedValueAsString)
+      if (!reset) {
+        setSearchTerms(selectedValueAsString)
+      }
       setResults({})
       setPageNum(0)
       setFetchType(FETCH_TYPE.NONE)
       setDoReset(false)
     }
-  }, [doReset, firstFilter, selectedValueAsString])
+  }, [doReset, firstFilter, selectedValueAsString, reset])
```

Comment:
This is confusing; we now have similarly named vars `doReset` and `reset` that mean different things; `reset` should probably be renamed to something clearer (e.g. `applySearchTerms` or so).